plugins/MauticCrmBundle/Controller/PublicController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic, Inc. All rights reserved
+ * @author      Mautic, Inc
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace MauticPlugin\MauticCrmBundle\Controller;
+
+use Mautic\CoreBundle\Controller\CommonController;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class class PublicController extends CommonController.
+ */
+class PublicController extends CommonController
+{
+    public function contactDataAction()
+    {
+        $content = $this->get('request')->getContent();
+        if (!empty($content)) {
+            $data = json_decode($content, true); // 2nd param to get as array
+        } else {
+            return new Response('ERROR');
+        }
+
+        $logger = $this->get('monolog.logger.mautic');
+
+        $integration       = 'Hubspot';
+        $integrationHelper = $this->factory->getHelper('integration');
+
+        $integrationObject = $integrationHelper->getIntegrationObject($integration);
+        foreach ($data as $info) {
+            $object = explode('.', $info['subscriptionType']);
+            $id     = $info['objectId'];
+
+            try {
+                switch ($object[0]) {
+                    case 'contact': $integrationObject->getContacts($id);
+                        break;
+                    case 'company':
+                        $this->factory->getLogger()->addError('creating company');
```

Comment:
Use `$logger->addError('creating company');` instead of factory. Is "creating company" really an error? Maybe it's forgotten debug code?